pkg/proxy/proxy_community_test.go

```diff
@@ -0,0 +1,374 @@
+				ExpectedProxyResponseStatus: http.StatusForbidden,
+				RequestPath:                 podsRequestURL("alice-private"),
+				ExpectedResponse:            "user access is forbidden: user access is forbidden",
+			},
+		}
+
+		for k, tc := range tests {
+			s.Run(k, func() {
+
+				// given
+				fakeApp.Err = nil
+
+				testServer.Config.Handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					w.Header().Set("Content-Type", "application/json")
+					// Set the Access-Control-Allow-Origin header to make sure it's overridden by the proxy response modifier
+					w.Header().Set("Access-Control-Allow-Origin", "dummy")
+					w.WriteHeader(http.StatusOK)
+					_, err := w.Write([]byte("my response"))
+					require.NoError(s.T(), err)
+					for hk, hv := range tc.ExpectedAPIServerRequestHeaders {
+						require.Len(s.T(), r.Header.Values(hk), len(hv))
+						for i := range hv {
+							assert.Equal(s.T(), hv[i], r.Header.Values(hk)[i], "header %s", hk)
+						}
+					}
+				})
+				fakeApp.SignupServiceMock = fake.NewSignupService(
+					fake.Signup(smith.String(), &signup.Signup{
+						Name:              "smith",
+						APIEndpoint:       testServer.URL,
+						ClusterName:       "member-2",
+						CompliantUsername: "smith",
+						Username:          "smith@",
+						Status: signup.Status{
+							Ready: true,
+						},
+					}),
+					fake.Signup(alice.String(), &signup.Signup{
+						Name:              "alice",
+						APIEndpoint:       testServer.URL,
+						ClusterName:       "member-2",
+						CompliantUsername: "alice",
+						Username:          "alice@",
+						Status: signup.Status{
+							Ready: true,
+						},
+					}),
+					fake.Signup(john.String(), &signup.Signup{
+						Name:              "john",
+						CompliantUsername: "john",
+						Username:          "john@",
+						Status: signup.Status{
+							Ready: false,
+						},
+					}),
+					fake.Signup(eve.String(), &signup.Signup{
+						Name:              "eve",
+						CompliantUsername: "eve",
+						Username:          "eve@",
+						Status: signup.Status{
+							Ready:  false,
+							Reason: toolchainv1alpha1.UserSignupUserBannedReason,
+						},
+					}),
+				)
+				s.Application.MockSignupService(fakeApp.SignupServiceMock)
+				inf := fake.NewFakeInformer()
+				inf.GetSpaceFunc = func(name string) (*toolchainv1alpha1.Space, error) {
+					switch name {
+					case "smith-community":
+						return fake.NewSpace("smith-community", "member-2", "smith"), nil
+					case "alice-private":
+						return fake.NewSpace("alice-private", "member-2", "alice"), nil
+					}
+					return nil, fmt.Errorf("space not found error")
+				}
+
+				sbSmithCommunitySmith := fake.NewSpaceBinding("smith-community-smith", "smith", "smith-community", "admin")
+				commSpacePublicViewer := fake.NewSpaceBinding("smith-community-publicviewer", toolchainv1alpha1.KubesawAuthenticatedUsername, "smith-community", "viewer")
```

Comment:
maybe this part also can be moved outside the test cases loop and the section after where you configure the informer listers and getters function if I'm not mistaken.

Reply:
good idea, done in 94e45a03b5b980db97572c7538eca8d1efd41880